vdr/doc/finder.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package doc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/vdr/types"
+)
+
+// ByServiceType returns a predicate that matches on service type
+// it only matches on DID Documents with a concrete endpoint (not starting with "did")
+func ByServiceType(serviceType string) types.Predicate {
+	return servicePredicate{serviceType: serviceType}
+}
```

Comment:
I'm wondering does it really make sense to only return DID documents with a concrete endpoint? Or should that be in a separate predicate?

Reply:
I had the same thoughts. I decided to make it explicit in the godoc, so whenever something else is needed, a simple refactor would suffice

Reply:
If we only care about DID documents with a concrete endpoint I think it would be okay like this. As long as we refactor it whenever we do want to return DID documents without a concrete endpoint